### PR TITLE
 Send datasets from projects/datasets activity

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds an export/share mechanism to be able to easily send edited datasets stored in the QField files directory to 3rd party apps or saved onto accessible storage locations.

Screencast:

https://user-images.githubusercontent.com/1728657/150809091-b8e5540f-9fde-4a75-b138-a6d890b20692.mp4

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for aad0dfc5610930562297397d4a6eda983a962ceb:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-97eb551ea402ba517f978a0ea0c3c332dc11bc3c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-97eb551ea402ba517f978a0ea0c3c332dc11bc3c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-97eb551ea402ba517f978a0ea0c3c332dc11bc3c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-97eb551ea402ba517f978a0ea0c3c332dc11bc3c-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6d7e3d42e55a5a5db0cf2417b1e7b0deb4589573:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-914e39dda0e2b186ecd2cb36dc5f505bcbe47485-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-914e39dda0e2b186ecd2cb36dc5f505bcbe47485-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-914e39dda0e2b186ecd2cb36dc5f505bcbe47485-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-914e39dda0e2b186ecd2cb36dc5f505bcbe47485-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f2b1080778ecb6d0ebbe882272796dffe8b63420:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-8284bd5294a31ff25a88961458cad67a0efbc908-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-8284bd5294a31ff25a88961458cad67a0efbc908-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-8284bd5294a31ff25a88961458cad67a0efbc908-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-8284bd5294a31ff25a88961458cad67a0efbc908-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9a7f8784a352f79765c18a6e9ac5cae545dc498f:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-f753d20de9fe6f7d1704e01f2d580217676e2e9c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-f753d20de9fe6f7d1704e01f2d580217676e2e9c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-f753d20de9fe6f7d1704e01f2d580217676e2e9c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-f753d20de9fe6f7d1704e01f2d580217676e2e9c-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 394a64b314d79b1e29aeabefa51fce7a752f5fad:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-187148c24f5029090e2545b5245682bd344b7971-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-187148c24f5029090e2545b5245682bd344b7971-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-187148c24f5029090e2545b5245682bd344b7971-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-187148c24f5029090e2545b5245682bd344b7971-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1158cc84f50d3c041d76244597174ae6a7a3d7c9:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-4b5f2333ebcea1d104cd7212a24093ab840bf76a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-4b5f2333ebcea1d104cd7212a24093ab840bf76a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-4b5f2333ebcea1d104cd7212a24093ab840bf76a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-4b5f2333ebcea1d104cd7212a24093ab840bf76a-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f0c8c3ad338af9e3caedd5ffe43bfe8185ead729:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-c75f85ae1c596bdc2a5d178763f8b1631a128e0b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-c75f85ae1c596bdc2a5d178763f8b1631a128e0b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-c75f85ae1c596bdc2a5d178763f8b1631a128e0b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-c75f85ae1c596bdc2a5d178763f8b1631a128e0b-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 127c4ded47e32ce823e3693d72df5903d81ab12f:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-94c1d80da7e03f72847210a1873a7d3560eb07ed-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-94c1d80da7e03f72847210a1873a7d3560eb07ed-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-94c1d80da7e03f72847210a1873a7d3560eb07ed-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-94c1d80da7e03f72847210a1873a7d3560eb07ed-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 387c2e2eca1ae76874b87892a3703103d57ac0f2:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-41de135ea77dabc29679808f8f6ee64fd472e88f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-41de135ea77dabc29679808f8f6ee64fd472e88f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-41de135ea77dabc29679808f8f6ee64fd472e88f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-41de135ea77dabc29679808f8f6ee64fd472e88f-x86.apk)

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 1]
```diff
@@ -79,54 +84,92 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
+                return super.onOptionsItemSelected(item);
         }
-        default:
-            return super.onOptionsItemSelected(item);
+    }
+
+    public boolean onMenuItemClick(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.send_to: {
+                final QFieldProjectListItem listItem =
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:132]: Should this be taken out of the switch?

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 2]
```diff
@@ -341,16 +386,59 @@ public void onRestart() {
...
+            }
+        }
+
+        currentPosition = position;
+        popupMenu.show();
+    }
+
     private void onItemClick(int position) {
-        Log.d(TAG, "onListItemClick ");
+        Log.d(TAG, "onItemClick ");
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:433]: I'm sometimes wondering how much of this "I am here" logging we need. No strong opinion.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1367b3ad1316e71d62eb6eda78f564bdd9ff976a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-dfd98343c08c982768f8a3fe4ce2126a2848e068-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-dfd98343c08c982768f8a3fe4ce2126a2848e068-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-dfd98343c08c982768f8a3fe4ce2126a2848e068-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-dfd98343c08c982768f8a3fe4ce2126a2848e068-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 08e27c940ad065632cc2d9dee3f71e0be875cf48:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-d43ceb55a715094bfa91bbf9aff07265dfeb4b47-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-d43ceb55a715094bfa91bbf9aff07265dfeb4b47-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-d43ceb55a715094bfa91bbf9aff07265dfeb4b47-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2431-d43ceb55a715094bfa91bbf9aff07265dfeb4b47-x86.apk)

**Pull request merged**